### PR TITLE
KTOR-7396 Migrate ktor-serialization tests to testApplication DSL

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
Client/Server, `ktor-serialization-*` modules

**Motivation**
[KTOR-7396](https://youtrack.jetbrains.com/issue/KTOR-7396/) Remove usage of deprecated `withTestApplication` DSL.

**Solution**
Migrated to `testApplication` DSL

### CODE REVIEW

File: ktor-shared/ktor-serialization/ktor-serialization-jackson/jvm/test/ServerJacksonTest.kt [hunk 1]
```diff
@@ -55,54 +56,52 @@ class ServerJacksonTest : AbstractServerSerializationTest() {
...
-        }.response.let { response ->
-            assertEquals(HttpStatusCode.OK, response.status())
-            assertNotNull(response.content)
-            assertEquals(listOf("""{"id":1,"title":"Hello, World!","unicode":"$uc"}"""), response.content!!.lines())
+        client.get("/") {
+            header(HttpHeaders.Accept, "application/json")
+            header(HttpHeaders.AcceptCharset, "UTF-16")
+        }.let { response ->
+            assertEquals(HttpStatusCode.OK, response.status)
+            assertEquals("""{"id":1,"title":"Hello, World!","unicode":"$uc"}""", response.bodyAsText())
```
Comment (osipxd) [inline ktor-shared/ktor-serialization/ktor-serialization-jackson/jvm/test/ServerJacksonTest.kt:64]: It looked like there was no need in `.lines()` call, so I removed it.

File: ktor-shared/ktor-serialization/ktor-serialization-tests/jvm/src/AbstractServerSerializationTest.kt [hunk 2]
```diff
@@ -68,103 +68,95 @@ public abstract class AbstractServerSerializationTest {
...
-            assertEquals(HttpStatusCode.OK, response.status())
-            assertNotNull(response.content)
-            assertEquals(listOf("""{"id":1,"title":"Hello, World!","unicode":"$uc"}"""), response.content!!.lines())
+        client.get("/map") {
+            header(HttpHeaders.Accept, "application/json")
+        }.let { response ->
+            assertEquals(HttpStatusCode.OK, response.status)
+            assertEquals("""{"id":1,"title":"Hello, World!","unicode":"$uc"}""", response.bodyAsText())
             val contentTypeText = assertNotNull(response.headers[HttpHeaders.ContentType])
             assertEquals(ContentType.Application.Json, ContentType.parse(contentTypeText))
```
Comment (osipxd) [inline ktor-shared/ktor-serialization/ktor-serialization-tests/jvm/src/AbstractServerSerializationTest.kt:150]: CI fails here. Investigating...

Comment (osipxd) [reply ktor-shared/ktor-serialization/ktor-serialization-tests/jvm/src/AbstractServerSerializationTest.kt]: Okay, I've changed the expected result for now. Any clues on why the actual result has changed with the new DSL are appreciated.

**Review by e5l — APPROVED** [review]
good job, lgtm

**Pull request merged**